src/app/item-popup/ItemSocketsWeapons.tsx

```diff
@@ -41,6 +41,14 @@ export default function ItemSocketsWeapons({ item, minimal, grid, onPlugClicked
 
   const handleSocketClick = (item: DimItem, socket: DimSocket, plug: DimPlug, hasMenu: boolean) => {
     if (hasMenu) {
+      if (socket.socketDefinition.socketTypeHash === 2218962841 && socket.plugSet) {
+        // Filter out unpluggable MW plugs
+        socket.plugSet.plugs = socket.plugSet.plugs.filter(
```

Comment:
You won't want to reassign this because it'll overwrite the global plugSet. You'll need to create a new plugs and assign it, which will be *involved*.

Reply:
I thought I had a clever way to do this with `_.once()`, but it also runs into this problem...